DistroLauncher/DistroLauncher.cpp

```diff
@@ -119,6 +119,12 @@ int wmain(int argc, wchar_t const *argv[])
         if (arguments.empty()) {
             hr = g_wslApi.WslLaunchInteractive(L"", false, &exitCode);
 
+            // Check exitCode to see if wsl.exe returned that it could not start the Linux process
+            // then prompt users for input so they can view the error message.
+            if (exitCode == UINT_MAX) {

```

Comment:
It's not safe to check exitCode if FAILED(hr).
```suggestion
            if (SUCCEEDED(hr) && exitCode == UINT_MAX) {

```